hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenAirdropHandlerTest.java

```diff
@@ -462,6 +467,62 @@ void handleAirdropMultipleTokensTransfers() {
         assertThat(relationToFungible.balance()).isEqualTo(1000L);
     }
 
+    @Test
+    void handleAirdropWithCustomFees() {
+        // setup all states
+        handlerTestBaseInternalSetUp(true);
+
+        // setup airdrop states
+        givenAssociatedReceiver(tokenReceiverId, nonFungibleTokenId);
+        givenAssociatedReceiver(tokenReceiverId, fungibleTokenId);
+        refreshReadableStores();
+        refreshWritableStores();
+        givenStoresAndConfig(handleContext);
+        given(handleContext.dispatchMetadata()).willReturn(DispatchMetadata.EMPTY_METADATA);
+        // mock record builder
+        tokenAirdropHandler = new TokenAirdropHandler(tokenAirdropValidator, validator);
+
+        final var fees = List.of(
+                withFixedFee(FixedFee.newBuilder().amount(66).build())
+                //                ,withFractionalFee(fractionalFee.copyBuilder().netOfTransfers(true).build())
+                );
+
+        // token with custom fees
+        var tokenWithNoCustomFees = fungibleToken.copyBuilder().customFees(fees).build();
+        //                fungibleToken.copyBuilder().customFees(Collections.emptyList()).build();
+
+        var nftWithNoCustomFees = nonFungibleToken
+                .copyBuilder()
+                .customFees(Collections.emptyList())
+                .build();
+        writableTokenStore.put(tokenWithNoCustomFees);
+        writableTokenStore.put(nftWithNoCustomFees);
+        given(storeFactory.writableStore(WritableTokenStore.class)).willReturn(writableTokenStore);
+        given(storeFactory.readableStore(ReadableTokenStore.class)).willReturn(writableTokenStore);
+
+        // TODO: I can see that the custom fees are being processed in
+        // TransferExecutor.chargeCustomFeeForAirdrops -> adjustHbarChangesStep.doIn
+        // but I don't see the amounts reduced by the fees. Does that mean the fees have been calculated
+        // but not actually applied yet?
```

Comment:
Intentional, I think? If so:
```suggestion
        // TODO: custom fees are processed in
        // TransferExecutor.chargeCustomFeeForAirdrops -> adjustHbarChangesStep.doIn
        // but the amounts don't seem to be reduced by the fees. Does that mean the fees have been calculated
        // but not actually applied yet?
```